src/field.cpp

```diff
@@ -1038,6 +1139,25 @@ bool map::process_fields_in_submap( submap *const current_submap,
                         }
 
                         // Consume adjacent fuel / terrain / webs to spread.
+                        // Allow raging fires (and only raging fires) to spread up
+                        // Spreading down is achieved by wrecking the walls/floor and then falling
+                        if( zlevels && cur->getFieldDensity() == 3 && p.z < OVERMAP_HEIGHT ) {
```

Comment:
I see it like this: fire burns the floor, the floor falls apart, burning pieces of the floor drop on the level below and burn there.

This will probably require marking some tiles as having a strong ceiling (not just using the floor from the tile above).

This doesn't work yet and currently nothing can burn down just to empty air - walls/floors that burn down leave ash in the air.